time.js

```diff
@@ -1,12 +1,169 @@
 const time = require('@js-joda/core');
 
 //openHAB uses a RFC DateTime string, js-joda defaults to the ISO version, this defaults RFC instead
 const rfcFormatter = time.DateTimeFormatter.ofPattern("uuuu-MM-dd'T'HH:mm:ss.SSS[xxxx][xxxxx]");
 const targetParse = time.ZonedDateTime.prototype.parse;
+
+/**
+ * When formatter is not supplied, uses an RFC DateTimeFormatter that can parse the format
+ * of java.time.ZonedDateTime.toString()
+ * @param {String} text 
+ * @param {time.DateTimeFormatter} formatter 
+ * @returns {time.ZonedDateTime}
+ * @throws {DateTimeParseException} when the text cannot be parsed
+ */
 time.ZonedDateTime.prototype.parse = function (text, formatter = rfcFormatter) {
     return targetParse(text, formatter);
 }
 
-module.exports = time;
+/**
+ * A passthrough function that can allow users to call `toZonedDateTime` on
+ * almost anything that can be converted to a time.ZonedDateTime without first
+ * checking to see if it's already a time.ZonedDateTime. 
+ * @returns {time.ZonedDateTime} this
+ */
+time.ZonedDateTime.prototype.toZonedDateTime = function () {
+    return this;
+}
+
+/**
+ * Change the date of the passed in dt to match the date. This preserves
+ * the hours and minutes even in cases where dt and date span a DST changeover.
+ * @param {time.ZonedDateTime} dt date time to move the date on
+ * @param {time.ZonedDateTime} date date time to move dt's date to
+ * @returns {time.ZonedDateTime} dt with date's date
+ */
+const changeDate = function (dt, date) {
+    return dt.withYear(date.year())
+             .withMonth(date.month())
+             .withDayOfMonth(date.dayOfMonth());
+}
+
+/**
+ * Moves the date portion of the date time to today, accounting for DST
+ * @returns {time.ZonedDateTime} this ZDT with today's date
+ */
+time.ZonedDateTime.prototype.toToday = function () {
+    return changeDate(this, time.ZonedDateTime.now());
+}
+
+/**
+ * Moves the date portion of the date time to tomorrow, accounting for DST
+ * @returns {time.ZonedDateTime} this ZDT with tomorrow's date
+ */
+time.ZonedDateTime.prototype.toTomorrow = function () {
+    return changeDate(this, time.ZonedDateTime.now().plusDays(1));
+}
+
+/**
+ * Moves the date portion of the date time to yesterday, accounting for DST
+ * @returns {time.ZonedDateTime} this ZDT with yesterday's date
+ */
+time.ZonedDateTime.prototype.toYesterday = function () {
+    return changeDate(this, time.ZonedDateTime.now().minusDays(1));
+}
+
+/**
+ * Compares this ZDT to see if it falls between start and end times,
+ * accounting for times that span midnight. start and end can be any type
+ * that has a `toZonedDateTime()` method but only the time portion is used.
+ * @param {*} start the starting time, anything that has a toZonedDateTime method
+ * @param {*} end the ending time, anything that has a toZonedDateTime method
+ * @returns {Boolean} true if this is between start and end
+ */
+time.ZonedDateTime.prototype.betweenTimes = function(start, end) {
```

Comment:
I initially called it `isBetween` at first but changed it because I find it to be misleading. 

The function only checks the if `this`'s time is between the passed in start and end times (accounting for a start that's before midnight and an end that's after). It uses the toTomorrow and toYesterday functions to position the dates to make that comparison possible. 

In essence it throws out the dates of start and end and only looks at the times. Consequently it's not really checking to see if `this` is between the start and end date times, which is what, to me at least `isBetween` implies.

Reply:
>In essence it throws out the dates of start and end and only looks at the times

Can you give me an example of how this would be used where the date would also not be in between?   I worry that adding methods to a ZDT that only operates on time, and ignores that date portion  may be too specialized to extend in our library and best left for users to do on their own (at this point you really want a Time object, not ZDT)

Reply:
errr, ignore my last comment about needing an example, i'm just reading your next comment which gives one.